HW0/P6/P6B.py

```diff
@@ -16,16 +16,25 @@ def burnTime(t):
 
     # Use a variety of wait times
     ratio = []
-    wait_time = []
+    wait_time = [pow(10, -k) for k in xrange(7)]
 
     for t in wait_time:
         # Compute jobs serially and in parallel
         # Use time.time() to compute the elapsed time for each
-        serialTime = 1
-        parallelTime = 1
+        serialTime = 0.
+        parallelTime = 0.
+
+        s_start = time.time()
+        for count in xrange(N):
+            burnTime(t)
+        serialTime = time.time() - s_start
+
+        p_start = time.time()
+        pool.map( burnTime(t), range(N) )
```

Comment:
I was curious about this, but couldn't come to a definitive answer. The expectation is that it should be a 4x improvement, but it appears that the sub-processes created by the Python multiprocessing library don't block when you call the sleep function. Instead, it appears to use some kind of threading mechanism to continue execution on the other tasks its assigned.

The implementation details for map are sparse, but it does split up the input array into chunks and sends it to the processes. I tried your suggestion, but it results in a similar graph on my system (`pool.map( burnTime, [t]*N )`). It should be equivalent to the original code in any case.

Any thoughts on this issue, @thouis?